ops/model.py

```diff
@@ -987,6 +993,38 @@ def location(self):
         return self._location
 
 
+class GoalState:
+    """Represents goal state of the Juju model."""
+    def __init__(self, backend):
+        self._backend = backend
+        self._goal_state = None
+
+    @property
+    def num_units(self) -> int:
+        """Return the number of expected peer units."""
+        goal_state = self.get_goal_state()
+        units = goal_state.get('units', {})
+        return len(units)
+
+    @property
+    def pending_units(self) -> int:
+        """Return the number of pending (non-Active) peer units."""
+        goal_state = self.get_goal_state()
+        units = goal_state.get('units', {})
+        return len([name
+                    for name, status in units.items()
+                    if status['status'].lower() != 'active'])
+
+    def get_goal_state(self) -> dict:
```

Comment:
I'm pretty sure this should be a private function, not part of the API of GoalState.